### PR TITLE
fix(argo-cd): Drop all `~` keys from redis-ha because they cause error on kustomize and cannot document via helm-docs

### PR DESCRIPTION
**Pull request by yu-croco**
This PR resolves https://github.com/argoproj/argo-helm/issues/1309 .
redis-ha subchart causes error with kustomize if we did not override `~` values (e.g.  [redis-ha. exporter.image](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/values.yaml#L787)). As a workaround, ~I added notes on values.yaml~. 🙋 
I removed the `~` keys from redis-ha .

---

Signed-off-by: yu-croco <yuki.kita22@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/main/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -784,9 +784,11 @@ redis-ha:
     enabled: true
     # -- Exporter image
     # @default -- `nil` (follows subchart default)
+    ## Note that default value (nil) causes error when using kustomize, so please make sure to override it as a workaround.
```
Comment (yu-croco) [inline charts/argo-cd/values.yaml:787]: Is it better to note on the top of [redis-ha](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/values.yaml#L776-L778) at once...? 🤔 (but maybe users would miss it ?)

**Review by mbevc1 — APPROVED** [review]
I cannot test this right now, but I'm happy with the notes. @mkilchhofer ?

**Comment by yu-croco** [comment]
@mkilchhofer 
Thank you for reviewing! I removed all of `~` keys. 🙋

**Comment by yu-croco** [comment]
> I'd rather prefer to partially revert https://github.com/argoproj/argo-helm/pull/1296
>
>-> With partial revert I mean: Drop all ~ keys from the values.yaml. I tried to document things via our helm-docs way which are not inside the values.yaml, but it seems that this is not possible with helm-docs

Hi @mkilchhofer , this is just a reminder. 🙋 
I removed all of `~` keys, so please check when you have time.

File: charts/argo-cd/Chart.yaml [hunk 2]
```diff
@@ -16,9 +16,9 @@ maintainers:
   - name: seanson
 dependencies:
   - name: redis-ha
-    version: 4.17.1
+    version: 4.17.2
```
[the remark this answers is not in the document]
Comment (yu-croco) [reply charts/argo-cd/Chart.yaml]: I'm sorry, I made a mistake. I pushed the right one... 🙏

**Comment by yu-croco** [comment]
Hi @mkilchhofer, 
Is there anything else I need to fix or any concerns at all? 👀 
*If it is not a good time to fix this, I will close this PR.

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Can we move the removed lines to the README.md.gotmpl ?

```diff
  | Key | Type | Default | Description |
  |-----|------|---------|-------------|
  {{- range .Values }}
   {{- if hasPrefix "redis-ha" .Key }}
  | {{ .Key }} | {{ .Type }} | {{ if .Default }}{{ .Default }}{{ else }}{{ .AutoDefault }}{{ end }} | {{ if .Description }}{{ .Description }}{{ else }}{{ .AutoDescription }}{{ end }} |
    {{- end }}
  {{- end }}
+ | redis-ha.exporter.image | string | `nil` (follows subchart default) | Exporter image |
+ | redis-ha.exporter.tag | string | `nil` (follows subchart default) | Exporter tag |
+ | redis-ha.haproxy.image.repository | string | `nil` (follows subchart default) | HAProxy Image Repository |
+ | redis-ha.haproxy.image.tag | string | `nil` (follows subchart default) | HAProxy Image Tag |
+ | redis-ha.image.repository | string | `nil` (follows subchart default) | Redis image repository |
```

Then we still have it documented and all users are happy ;-)

**Comment by yu-croco** [comment]
Thank you for reviewing! That's a good idea.
I fixed on https://github.com/argoproj/argo-helm/pull/1322/commits/8e87016fc12d8e25065dc4081f1a548dc0bdd834 . 🙋

File: charts/argo-cd/README.md.gotmpl [hunk 3]
```diff
@@ -300,6 +300,11 @@ If you want to use an existing Redis (eg. a managed service from a cloud provide
 | {{ .Key }} | {{ .Type }} | {{ if .Default }}{{ .Default }}{{ else }}{{ .AutoDefault }}{{ end }} | {{ if .Description }}{{ .Description }}{{ else }}{{ .AutoDescription }}{{ end }} |
   {{- end }}
 {{- end }}
+| redis-ha.exporter.image | string | `nil` (follows subchart default) | Exporter image |
+| redis-ha.exporter.tag | string | `nil` (follows subchart default) | Exporter tag |
+| redis-ha.haproxy.image.repository | string | `nil` (follows subchart default) | HAProxy Image Repository |
+| redis-ha.haproxy.image.tag | string | `nil` (follows subchart default) | HAProxy Image Tag |
+| redis-ha.image.repository | string | `nil` (follows subchart default) | Redis image repository |
```
Comment (mkilchhofer) [inline charts/argo-cd/README.md.gotmpl:307]: Can you move it around 12 lines up into the Redis HA section? ;)

Comment (yu-croco) [reply charts/argo-cd/README.md.gotmpl]: Oh, I moved them to wrong place...😇  I fixed it again.

**Review by mkilchhofer — APPROVED** [review]
LGTM. Thanks for fixing this @yu-croco 🙏

**Pull request merged**